src/POWERExecution.cpp

```diff
@@ -2645,7 +2645,13 @@ void POWERInterpreter::registerOperand(int proc, FetchedInstruction &FI, int idx
       llvm::Type *ty =
         FI.I.getOperand(idx)->getType();
       assert(ty->isPointerTy());
+#if LLVM_VERSION_MAJOR > 15
+      if(ty->isOpaquePointerTy()){
+	ty = llvm::PointerType::get(llvm::cast<llvm::PointerType>(ty)->getContext(), 0);
```

Comment:
What's even going on here? Why would making `ty` a pointer type work? It's not even (necessarily) the same size as the pointee!

When I did llvm15 support, my judgement was that supporting opaque pointers in POWER/ARM would be a significant effort, and since I was not so familiar with the implementation, I decided to let it become deprecated. I suggest just excluding POWER/ARM from the build completely when using a version of llvm that does not have typed pointers anymore. We'll probably support llvm 15 for a long time, so I doubt this implementation will be research-relevant anymore once we finally completely drop it.